Sources/CoreLocationMiddleware/CoreLocationMiddleware.swift

```diff
@@ -1,50 +1,62 @@
-    case unknown
-    case notAuthorized
-    case authorized(lastPosition: CLLocation?)
+public struct LocationState: Equatable {
+    var authzType: AuthzType
+    var authzStatus: CLAuthorizationStatus
+    var location: CLLocation
+    
+    public init(authzType: AuthzType = .whenInUse,
+                authzStatus: CLAuthorizationStatus,
+                location: CLLocation) {
+        self.authzType = authzType
+        self.authzStatus = authzStatus
+        self.location = location
+    }
 }
 
-public enum LocationAction {
+public enum LocationAction: Equatable {
+    // Input
     case startMonitoring
     case stopMonitoring
+    case requestAuthorizationStatus
+    case requestAuthorizationType
+    case requestPosition
+    case requestDeviceCapabilities
+    // Output
     case gotPosition(CLLocation)
-    case authorized
-    case unauthorized
-    case authorizationUnknown
-    case receiveError(Error)
+    case gotAuthzStatus(CLAuthorizationStatus)
+    case gotDeviceCapabilities(DeviceCapabilities)
+    case receiveError(CLError)
 }
 
-extension LocationAction: Equatable {
-    public static func == (lhs: LocationAction, rhs: LocationAction) -> Bool {
-        switch (lhs, rhs) {
-        case let (.receiveError(x), .receiveError(y)): return x.localizedDescription == y.localizedDescription
-        case let (.gotPosition(x), .gotPosition(y)): return x == y
-        case (.startMonitoring, .startMonitoring), (.stopMonitoring, .stopMonitoring), (.authorized, .authorized), (.unauthorized, .unauthorized), (.authorizationUnknown, .authorizationUnknown) : return true
-        default:
-            return false
-        }
-    }
+public enum AuthzType: Equatable {
+    case whenInUse
+    case always
 }
 
+public struct DeviceCapabilities: Equatable {
+    public let isSignificantLocationChangeAvailable: Bool
+    public let isHeadingAvailable: Bool
+    public let isRegionMonitoringAvailable: Bool
+    public let isRangingAvailable: Bool
+    public let isLocationServiceAvailable: Bool
+}
+
+
 let locationReducer = Reducer<LocationAction, LocationState> { action, state in
     var state = state
     switch action {
-    case .authorized:
-        if case .authorized = state { return state }
-        state = .authorized(lastPosition: nil)
-    case .startMonitoring, .stopMonitoring:
-        break
-    case let .gotPosition(position):
-        state = .authorized(lastPosition: position)
-    case .unauthorized:
-        state = .notAuthorized
-    case .authorizationUnknown:
-        state = .unknown
-    case .receiveError:
+    case .startMonitoring,
+         .stopMonitoring,
+         .requestAuthorizationStatus,
+         .requestAuthorizationType,
+         .requestPosition,
+         .requestDeviceCapabilities:
         break
+    case let .gotAuthzStatus(status): state.authzStatus = status
+    case let .gotPosition(position): state.location = position
+    case .gotDeviceCapabilities, .receiveError : break
```

Comment:
Ah ! ok, so this is one of the things I don't quite understand yet with Middleware.📝🧐 (I need to take some notes here)

So, if I am capturing this correctly, are you saying : receive the getDeviceCapability in Input, **but** instead of returning the values via the (output) action gotDeviceCapabilities, then just populate the Middleware's LocationState ?
If so how does the change to the Middleware's state get reflected back to the AppState ? Is it by lifting it back up from the Middleware's `stateMap` ?

```
        stateMap: { globalState -> LocationState in
            return LocationState(
                authzType: globalState.authorizationType,
                authzStatus: globalState.authorizationStatus,
                location: globalState.location
            )
        }
```

Or you're just suggesting I add those in the Middleware's state just so we have them ?

Reply:
See #8